lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/StackPresenter.java

```diff
@@ -186,35 +191,40 @@ private void applyTopBarOptions(Options options, StackController stack, ViewCont
         }
 
         topBar.setTitleHeight(topBarOptions.title.height.get(UiUtils.getTopBarHeightDp(activity)));
-        topBar.setTitle(topBarOptions.title.text.get(""));
         topBar.setTitleTopMargin(topBarOptions.title.topMargin.get(0));
 
+        //check for set component
         if (topBarOptions.title.component.hasValue()) {
             if (titleControllers.containsKey(component)) {
                 topBarController.setTitleComponent(titleControllers.get(component));
             } else {
                 TitleBarReactViewController controller = new TitleBarReactViewController(activity, titleViewCreator, topBarOptions.title.component);
                 controller.setWaitForRender(topBarOptions.title.component.waitForRender);
                 titleControllers.put(component, controller);
-                controller.getView().setLayoutParams(getComponentLayoutParams(topBarOptions.title.component));
                 topBarController.setTitleComponent(controller);
             }
+            topBarController.alignTitleComponent(topBarOptions.title.component.alignment);
+        } else {
+            //check for text title
+            String titleText = topBarOptions.title.text.get("");
+            topBar.setTitle(titleText);
+            topBar.setTitleFontSize(topBarOptions.title.fontSize.get(defaultTitleFontSize));
+            topBar.setTitleTextColor(topBarOptions.title.color.get(DEFAULT_TITLE_COLOR));
+            topBar.setTitleTypeface(typefaceLoader, topBarOptions.title.font);
+            topBar.setTitleAlignment(topBarOptions.title.alignment);
+
+            //check for text subtitle
+            String subTitleText = topBarOptions.subtitle.text.get("");
+            topBar.setSubtitle(subTitleText);
+            topBar.setSubtitleFontSize(topBarOptions.subtitle.fontSize.get(defaultSubtitleFontSize));
+            topBar.setSubtitleColor(topBarOptions.subtitle.color.get(DEFAULT_SUBTITLE_COLOR));
+            topBar.setSubtitleTypeface(typefaceLoader, topBarOptions.subtitle.font);
+            topBar.setSubtitleAlignment(topBarOptions.subtitle.alignment);
```

Comment:
Feels like we can split it into two functions, wdyt?

Reply:
it can be split for sure, but I think adding functions just for naming (called in one place) is a bit of an overhead to achieve good readability, however, comments will explain a section of the code without changing the code, and in terms of maintainability its the same for functions, if you changed the responsibility of the function you might want to change the name too, hence, changing comment and a function name has the same work cost, and we get the same readability without touching the code flow unless the code is a duplication that needs to be addressed, and in this code, I would suggest using comments instead of functions.

Reply:
even without comments, the code is self-explaining by itself.